scene/gui/tree.cpp

```diff
@@ -709,12 +738,37 @@ TreeItem *TreeItem::get_next_visible(bool p_wrap) {
 	return current;
 }
 
+TreeItem *TreeItem::get_next_visible(bool p_wrap) {
+	TreeItem *loop = this;
+	TreeItem *next = this->_get_next_visible(p_wrap);
+	while (next && !next->is_visible()) {
+		next = next->_get_next_visible(p_wrap);
+		if (next == loop) {
+			// Check that we haven't looped all the way around to the start.
+			next = nullptr;
+			break;
+		}
+	}
+	return next;
+}
+
 TreeItem *TreeItem::get_child(int p_idx) {
 	_create_children_cache();
 	ERR_FAIL_INDEX_V(p_idx, children_cache.size(), nullptr);
 	return children_cache.get(p_idx);
 }
 
+int TreeItem::get_visible_child_count() {
```

Comment:
I didn't publicly expose this function, but if it might be useful then I can add it. I can see it having use, but it's also not as efficient for determining whether to conditionally hide something based on whether it has any visible children as simply counting how many children are not hidden as you hide them to avoid doing a second loop.
It still may be useful though in other situations...